### PR TITLE
steamback: init at 0.3.6

### PR DESCRIPTION
**Pull request by AngryAnt**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AngryAnt** [comment]
Needs `async-tkinter-loop` and `sv-ttk` split out into individual packages. Probably also a better solution for `TK_LIBRARY`.

**Comment by AngryAnt** [comment]
Blocked by pending PRs:
 - https://github.com/NixOS/nixpkgs/pull/235802
 - https://github.com/NixOS/nixpkgs/pull/235803

**Comment by AngryAnt** [comment]
Build issue pending resolution of https://github.com/NixOS/nixpkgs/issues/238990

File: pkgs/tools/games/steamback/default.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  pythonRelaxDeps = [
+    "async-tkinter-loop"
+    "platformdirs"
+    "Pillow"
+  ];
+
+  makeWrapperArgs = [
+    ''--prefix TK_LIBRARY : "${pkgs.tk}/lib/tk8.6"''
+  ];
```
Comment (wegank) [inline pkgs/tools/games/steamback/default.nix:43]:
```suggestion
```

Fixed two weeks ago.

Comment (AngryAnt) [reply pkgs/tools/games/steamback/default.nix]: Ah excellent! This PR has been a while underway and I did not notice. Thanks! :)

**Comment by AngryAnt** [comment]
@ofborg build steamback

**Comment by AngryAnt** [comment]
@NixOS/ofborg build steamback

File: pkgs/tools/games/steamback/default.nix [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  nativeBuildInputs = with python311.pkgs; [
+    pythonRelaxDepsHook
+  ];
+
+  buildInputs = with python311.pkgs; [
+    setuptools
+    pillow
+  ];
+
+  propagatedBuildInputs = with python311.pkgs; with python311Packages; [
```
Comment (wegank) [inline pkgs/tools/games/steamback/default.nix:27]:
```suggestion
  propagatedBuildInputs = with python311.pkgs; [
```

File: pkgs/tools/games/steamback/default.nix [hunk 3]
```diff
@@ -0,0 +1,56 @@
+{ lib
+, pkgs
+, python311
+, fetchPypi
+, python311Packages
```
Comment (wegank) [inline pkgs/tools/games/steamback/default.nix:5]:
```suggestion
```

**Comment by AngryAnt** [comment]
Excellent. I'm listening in on the `tkinter` issue thread and will push a shift from `python311` to `python3` when it is resolved.

**Review by wegank — APPROVED** [review]

File: pkgs/tools/games/steamback/default.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
+{ lib
+, pkgs
```
Comment (wegank) [inline pkgs/tools/games/steamback/default.nix:2]:
```suggestion
```

**Review by Janik-Haag — APPROVED** [review]
LGTM

File: pkgs/tools/games/steamback/default.nix [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  pythonRelaxDeps = [
+    "async-tkinter-loop"
+    "platformdirs"
+    "Pillow"
+  ];
+
+  checkPhase = ''
+    runHook preCheck
+
+    $out/bin/${pname} --help
```
Comment (SuperSandro2000) [inline pkgs/tools/games/steamback/default.nix:50]: Thats not cross friendly and I think we have some helper function for this

File: pkgs/tools/games/steamback/default.nix [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    inherit pname version;
+    hash = "sha256-hvMPSxIfwwQqo80JCpYhcbVY4kXs5jWtjjafVSMrw6o=";
+  };
+
+  nativeBuildInputs = [
+    pythonRelaxDepsHook
+  ];
+
+  buildInputs = [
+    setuptools
```
Comment (SuperSandro2000) [inline pkgs/tools/games/steamback/default.nix:29]: setuptools probably goes to nativeBuildInputs like pillow if it is run on the build machine

**Pull request merged**